packages/synapse-react-client/src/components/ReleaseCard/ReleaseCardUtils.ts

```diff
@@ -0,0 +1,146 @@
+import { Query } from '@sage-bionetworks/synapse-types'
+import dayjs from 'dayjs'
+import { formatDate } from '../../utils/functions/DateFormatter'
+import { ReleaseCardSchema } from './ReleaseCard'
+import {
+  ButtonToExplorePageConfig,
+  ReleaseCardStat,
+  StatConfig,
+} from './ReleaseCardTypes'
+
+const MISSING_STAT_DEFAULT_VALUE = 'Unknown'
+
+const formatNumber = (num: number) => {
+  if (num < 1_000_000) return num.toLocaleString()
+  const numberFormatter = new Intl.NumberFormat([], {
+    minimumFractionDigits: 1,
+    maximumFractionDigits: 2,
+    notation: 'compact',
+    localeMatcher: 'best fit',
+  })
+  return numberFormatter.format(num)
+}
+
+const getValueFromData = (
+  schema: ReleaseCardSchema,
+  data: (string | null)[],
+  columnName: string,
+) => {
+  const columnIndex = schema[columnName]
+  let columnValue
+  if (columnIndex === undefined) {
+    console.error(`Could not find column named ${columnName}`)
+    columnValue = null
+  } else {
+    columnValue = data[columnIndex]
+  }
+  return columnValue
+}
+
+const formatReleaseStats = (
+  schema: ReleaseCardSchema,
+  data: (string | null)[],
+  statsConfig: StatConfig[],
+) => {
+  return statsConfig.map(statConfig => {
+    const { columnName, label } = statConfig
+    const statString = getValueFromData(schema, data, columnName)
+    const cardStat: ReleaseCardStat = {
+      label: label,
+      value: statString
+        ? formatNumber(Number(statString))
+        : MISSING_STAT_DEFAULT_VALUE,
+    }
+    return cardStat
+  })
+}
+
+const formatReleaseDate = (
+  schema: ReleaseCardSchema,
+  data: (string | null)[],
+) => {
+  const releaseDateString = data[schema.releaseDate]
+  const cardStat: ReleaseCardStat = {
+    label: 'Date of release',
+    value: releaseDateString
+      ? formatDate(dayjs(Number(releaseDateString)), 'MM/DD/YYYY')
+      : MISSING_STAT_DEFAULT_VALUE,
+  }
+  return cardStat
+}
+
+export const formatReleaseCardData = (
+  schema: ReleaseCardSchema,
+  data: (string | null)[],
+  statsConfig: StatConfig[],
+) => {
+  return {
+    releaseVersion: data[schema.releaseVersion],
+    releaseEntity: data[schema.releaseEntity],
+    releaseDate: formatReleaseDate(schema, data),
+    stats: formatReleaseStats(schema, data, statsConfig),
+  }
+}
+
+const createButtonToExploreDataQueryString = (
+  schema: ReleaseCardSchema,
+  data: (string | null)[],
+  buttonToExploreDataConfig?: ButtonToExplorePageConfig,
+) => {
+  if (!buttonToExploreDataConfig) return null
+
+  const {
+    exploreDataSql,
+    exploreDataFacetColumnName,
+    sourceDataFacetValueColumnName,
+  } = buttonToExploreDataConfig
+  if (
+    exploreDataSql &&
+    exploreDataFacetColumnName &&
+    sourceDataFacetValueColumnName
+  ) {
+    const facetValue = getValueFromData(
+      schema,
+      data,
+      sourceDataFacetValueColumnName,
+    )
+    // column not found in source table or cell did not have value in source table
+    if (facetValue === null) return null
```

Comment:
For the GENIE data releases, the source table will have a null value in the facet value column, so that the button will redirect to `/explore/files/genie` without selected facets. Would you include that context in the warning, i.e. that the facet value may be null by design?